Source/WTF/wtf/text/TextBreakIterator.h

```diff
@@ -181,13 +182,13 @@ class CachedTextBreakIterator {
     WTF_MAKE_FAST_ALLOCATED;
 public:
     CachedTextBreakIterator(StringView string, const UChar* priorContext, unsigned priorContextLength, TextBreakIterator::Mode mode, const AtomString& locale, TextBreakIterator::ContentAnalysis contentAnalysis = TextBreakIterator::ContentAnalysis::Mechanical)
-        : m_backing(TextBreakIteratorCache::singleton().take(string, priorContext, priorContextLength, mode, contentAnalysis, locale))
+        : m_backing(isMainThread() ? TextBreakIteratorCache::singleton().take(string, priorContext, priorContextLength, mode, contentAnalysis, locale) : TextBreakIterator(string, priorContext, priorContextLength, mode, contentAnalysis, locale))
```

Comment:
Don't we want `isMainRunLoop`?

Reply:
I believe `isMainRunLoop` is for WebKit code, `isMainThread` is for WebCore and WTF.